node-notes-cli/command-file.js

```diff
@@ -11,109 +11,125 @@ if (process.argv.length > 5) {
- * and updates the nextID property of dat.json and rewrites the file.
- * Added bug fix to make sure user inputs a new note when using create command.
- * @returns returns new data.json text
+ * Logs an objects notes.
  */
+async function readAndReport() {
+  try {
+    const theObject = await read();
+    for (const [key, value] of Object.entries(theObject.notes)) {
+      console.log(`${key}: ${value}`);
+    }
+  } catch (err) {
+    console.error('Read and report file failed:', err);
+  }
+}
 
-async function create() {
+/**
+ * Creates a new notes entry in the object and writes the data.json file.
+ * @param A new note to be added.
+ */
+async function create(newNote) {
   try {
-    if (!input1) {
+    if (!newNote) {
       console.error('This command requires an additional argument to create a note.');
       process.exit(1);
     }
     const objectParsed = await read();
     const currentID = objectParsed.nextId;
-    objectParsed.notes[currentID] = input1;
+    objectParsed.notes[currentID] = newNote;
     objectParsed.nextId++;
-    const newFile = await fs.writeFile('data.json', JSON.stringify(objectParsed, null, 2));
-    return newFile;
+    await writeToFile(objectParsed);
+  } catch (err) {
+    console.error('Could not create a new note:', err);
+  }
+}
+
+/**
+ * Converts an object to JSON and writes it to data.json.
+ * @param object to be converted to JSON
+ */
+async function writeToFile(content) {
+  try {
+    const newJSON = JSON.stringify(content, null, 2);
+    fs.writeFile('data.json', newJSON);
   } catch (err) {
-    console.error('Something went wrong', err);
+    console.error('File failed to write:', err);
   }
 }
+
 /**
- * The update function takes the read data.json and updates the users chosen note
- * then rewrites the file. Added conditional to make sure entry number exists.
- * Removed bug that would delete the note if no update argument was passed.
- * @returns updated data.json text
+ * Updates the users chosen note.
+ * @param noteID to be replaced by newNote made by user
  */
-async function update() {
+async function update(noteID, newNote) {
   try {
-    if (!input2) {
+    if (!newNote) {
       console.error('This command requires an additional argument to update the note.');
       process.exit(1);
     }
     const objectParsed = await read();
-    if (!Object.keys(objectParsed.notes).includes(input1)) {
-      console.error('Not a entry number:', input1);
+    if (!Object.keys(objectParsed.notes).includes(noteID)) {
+      console.error('Not a entry number:', noteID);
       process.exit(1);
     }
-    const currentID = input1;
-    objectParsed.notes[currentID] = input2;
-    const newFile = await fs.writeFile('data.json', JSON.stringify(objectParsed, null, 2));
-    return newFile;
+    objectParsed.notes[noteID] = newNote;
+    await writeToFile(objectParsed);
   } catch (err) {
-    console.error('Something went wrong', err);
+    console.error('Update failed:', err);
   }
 }
 /**
```

Comment:
There should be a blank line between the previous function and the next JSDoc comment. You do that elsewhere, so this must just be a simple typo.